### ISSUE TITLE
Store data as serialized array

### ISSUE THREAD

**Issue by markhowellsmead** [mhmli/mhm-flickrsettings#3]
This will break previous saved data; that will need to be handled when reading out data stored with a former version of the plugin.

**Comment by grappler** [comment]
You could solve the fetching of the settings with:

``` php
add_filter( 'option_flickr_key', function ( $value ) {
    $settings = get_option( 'mhm_flickr_settings'):
    if ( isset( $settings['flickr_key'] ) ) {
        $value = $settings['flickr_key'];
    }
    return $value;
}
```

Change to only `register_setting( 'media', 'mhm_flickr_settings', 'esc_attr' );` and change the name in the three settings `name="mhm_flickr_settings[flickr_key]" id="flickr_key"`

**Comment by markhowellsmead** [comment]
Thanks @grappler! Took a while to get round to it, but these changes are implemented in v1.2.0. (Here and in the WordPress Plugin Repository.)
